<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>873</width>
    <height>681</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>70</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>PIDtester</string>
    </property>
   </widget>
   <widget class="QLabel" name="text_nowpose">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>190</y>
      <width>151</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string>now_pose:
x:0
y:0
w:0</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="show_msgs">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>50</y>
      <width>311</width>
      <height>101</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>30</y>
      <width>111</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>show_msgs</string>
    </property>
   </widget>
   <widget class="QPushButton" name="button_set_target">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>100</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>set_target</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>270</y>
      <width>271</width>
      <height>128</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="text_kp">
           <property name="text">
            <string>kp:00</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="set_kp"/>
         </item>
         <item>
          <widget class="QPushButton" name="enble_kp">
           <property name="text">
            <string>enable_kp</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="text_ki">
           <property name="text">
            <string>ki:00</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="set_ki"/>
         </item>
         <item>
          <widget class="QPushButton" name="enbale_ki">
           <property name="text">
            <string>enable_ki</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="text_kd">
           <property name="text">
            <string>kd:00</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="set_kd"/>
         </item>
         <item>
          <widget class="QPushButton" name="enable_kd">
           <property name="text">
            <string>enable_kd</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="enable_pid_run">
       <property name="text">
        <string>enable_pid_run</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>90</y>
      <width>301</width>
      <height>95</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="text_target_x">
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>target_x:x</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="set_target_x"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="text_target_y">
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>target_x:y</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="set_target_y"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="text_target_theta">
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>target_theta:theta</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="set_target_theta"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="global_clean">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>220</y>
      <width>91</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>global_clean</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pass_clean">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>280</y>
      <width>91</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>pause_clean</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>411</width>
      <height>54</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="text_target_w_2">
         <property name="text">
          <string>IIP:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="set_rpc_IP">
         <property name="text">
          <string>192.168.1.100</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="connect_rpc">
         <property name="text">
          <string>con_rpc</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QLabel" name="text_target_w_3">
         <property name="text">
          <string>port:50001</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="CoordinateMap" name="widget_map" native="true">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>230</y>
      <width>400</width>
      <height>400</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>400</width>
      <height>400</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>400</width>
      <height>400</height>
     </size>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>190</y>
      <width>301</width>
      <height>27</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <item>
      <widget class="QLineEdit" name="set_map_x_range"/>
     </item>
     <item>
      <widget class="QLabel" name="show_map_x_range">
       <property name="text">
        <string>range_x:4000mm</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>410</y>
      <width>191</width>
      <height>27</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_10">
     <item>
      <widget class="QLabel" name="text_pid_v">
       <property name="text">
        <string>pid_v:0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="set_pid_v"/>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget_2">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>160</y>
      <width>271</width>
      <height>27</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_11">
     <item>
      <widget class="QLabel" name="text_allow_erro">
       <property name="text">
        <string>近点允许误差0mm</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="set_allow_erro"/>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="clean_history">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>380</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>清除轨迹点
</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>873</width>
     <height>28</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CoordinateMap</class>
   <extends>QWidget</extends>
   <header location="global">coordinatemap.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
